classes/mail/mailables/ReviewRemind.php

```diff
@@ -49,60 +46,22 @@ class ReviewRemind extends Mailable
     protected Context $context;
     protected ReviewAssignment $reviewAssignment;
 
-    public function __construct(Context $context, PKPSubmission $submission, ReviewAssignment $reviewAssignment)
+    public function __construct(Context $context, Submission $submission, ReviewAssignment $reviewAssignment)
     {
         parent::__construct(func_get_args());
+
         $this->context = $context;
         $this->reviewAssignment = $reviewAssignment;
     }
 
-    /*
-     * Override reviewAssignmentUrl template variable if one-click reviewer access is enabled and add passwordResetUrl
-     */
-    public function recipients(User $recipient, ?string $locale = null): Mailable
-    {
-        $this->traitRecipients([$recipient], $locale);
-        $request = Application::get()->getRequest();
-        $dispatcher = $request->getDispatcher();
-
-        if ($this->context->getData('reviewerAccessKeysEnabled')) {
-            $accessKeyManager = new AccessKeyManager();
-            $expiryDays = ($this->context->getData('numWeeksPerReview') + 4) * 7;
-            $accessKey = $accessKeyManager->createKey(
-                $this->context->getId(),
-                $recipient->getId(),
-                $this->reviewAssignment->getId(), $expiryDays
-            );
-            $reviewUrlArgs = [
-                'submissionId' => $this->reviewAssignment->getSubmissionId(),
-                'reviewId' => $this->reviewAssignment->getId(),
-                'key' => $accessKey,
-            ];
-
-            $this->viewData[ReviewAssignmentEmailVariable::REVIEW_ASSIGNMENT_URL] =
-                $dispatcher->url(
-                    $request,
-                    Application::ROUTE_PAGE,
-                    $this->context->getData('urlPath'),
-                    'reviewer',
-                    'submission',
-                    null,
-                    $reviewUrlArgs
-                );
-        }
-
-        // Old REVIEW_REMIND template contains additional variable not supplied by _Variable classes
-        $this->setPasswordResetUrl($recipient, $this->context->getData('urlPath'));
-
-        return $this;
-    }
-
     /**
-     * @copydoc Mailable::getDataDescriptions()
+     * Override the setData method to add the one-click access
+     * URL for the reviewer
      */
-    public static function getDataDescriptions(): array
+    public function setData(?string $locale = null): void
```

Comment:
Yes, the function `addData()` will add information to the `viewData` property. However, when the email template variables are compiled, any matching keys will be overwritten by the email template variables.

Since this variable uses the same key as a email template variable, `reviewAssignmentUrl`, it has to be set _after_ the email template variables are rendered.